[TP-198470] Update to Nanoc v4

ssawyer1024 opened the pull request:
- **[TP-198470] Update gemfiles to "modern" ruby/gems and nanoc4**
- **[TP-198470] Minimal changes for nanoc v3 -> v4 migration to get the site up and running.**

AndrewHYi reviewed (commented):
- `.ruby-version` is set to `2.4.1.`, should be upgrade that to ruby 3?
- The version specified in `.ruby-version` wasn't being picked up by asdf. I had to set [legacy mode true ](https://github.com/asdf-vm/asdf-ruby?tab=readme-ov-file#migrating-from-another-ruby-version-manager). Should we switch to using `.tool-versions`?
- This project was bundled with an old bundler version.  Should we upudate bundler to v. 2? `bundle update --bundler`

ssawyer1024 commented:
> * `.ruby-version` is set to `2.4.1.`, should be upgrade that to ruby 3?
Fixed.

> * The version specified in `.ruby-version` wasn't being picked up by asdf. I had to set [legacy mode true ](https://github.com/asdf-vm/asdf-ruby?tab=readme-ov-file#migrating-from-another-ruby-version-manager). Should we switch to using `.tool-versions`?
Added.

> * This project was bundled with an old bundler version.  Should we upudate bundler to v. 2? `bundle update --bundler`
Fixed.

ssawyer1024 commented:
@AndrewHYi It should work properly now. Combination of not quite getting /static right with being too aggressive moving away from legacy (which I've reverted). If we want to do more with a micro-site perhaps we use something other than nanoc (Roda? 😁 )...
